parser/src/cfg/mod.rs

```diff
@@ -1068,13 +1116,49 @@ fn parse_layer_indexes(exprs: &[SpannedLayerExprs], expected_len: usize) -> Resu
                         )
                     })?
                     .to_owned();
-                if list.len() != 1 {
-                    bail_expr!(
-                        layer_expr,
-                        "layer name after {DEFLAYER_MAPPED} must be a string within one pair of parentheses"
-                    );
+                let mut list = list.iter();
+                let name = list.next().ok_or_else(|| {anyhow_expr!(layer_expr,"layer name after {DEFLAYER_MAPPED} must be a string within one pair of parentheses")})?
+                    .atom(None).ok_or_else(|| {anyhow_expr!(layer_expr, "layer name after {DEFLAYER_MAPPED} must be a string within one pair of parentheses")})?;
+
+                let mut icon = "";
+                // add hashmap for future options, currently only parse icons
+                let mut layer_opts: HashMap<String, String> = HashMap::default();
+                while let Some(key_expr) = list.next() {
+                    // Read k-v pairs from the configuration
+                    let opt_key = key_expr.atom(None).ok_or_else(|| {anyhow_expr!(&key_expr, "No lists are allowed in {DEFLAYER_MAPPED} options")})
```

Comment:
Please make sure to add tests for this code

Reply:
added